src/System.Linq/tests/ShortCircuitingTests.cs

```diff
@@ -66,7 +69,10 @@ public void MinDoubleDoesntCheckAll()
             var tracker = new TrackingEnumerable(10);
             var source = tracker.Select(i => i == 5 ? double.NaN : (double)i);
             Assert.True(double.IsNaN(source.Min()));
-            Assert.Equal(5, tracker.Moves);
+
+            // .NET core shortcircuits as an optimization.
+            // See https://github.com/dotnet/corefx/pull/2350.
+            Assert.Equal(PlatformDetection.IsFullFramework ? 10 : 5, tracker.Moves);
```

Comment:
Applies to other diffs in this file.

---
In reply to: [110053838](https://github.com/dotnet/corefx/pull/17877#discussion_r110053838) [](ancestors = 110053838,109671547)